.github/workflows/translate.yml

```diff
@@ -0,0 +1,83 @@
+name: "Update translations"
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 3 * * *"
+  push:
+    branches:
+      - master
+    
+permissions: {}
+
+defaults:
+  run:
+    shell: pwsh
+
+jobs:
+  update-translations:
+    permissions:
+      contents: write # for git push
+    runs-on: ubuntu-latest
+
+    env:
+      TX_TOKEN: ${{ secrets.TX_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          ref: ${{ github.ref }}
+
+      - name: l10n-push
+        if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+        # we need to use a different docker image for those two, this appears to be a bit tricky with github actions out of the box
+        run:  bash .github/workflows/run-in-docker.sh owncloudci/transifex:latest "cd ownCloud/Resources/en.lproj && tx push -s --skip"
+
+      - name: l10n-pull
+        run: bash .github/workflows/run-in-docker.sh owncloudci/transifex:latest "cd ownCloud/Resources/ && tx pull --force --skip --all"
+
+      - name: update-repo-before-commit
+        run: |
+          git status
+          git add ownCloud/Resources/*.strings
+          git status
+          git stash
+          git pull --ff-only origin
+
+      - name: commit and push
+        if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+        run: |
+          install -d -m 0700 ~/.ssh
+          Set-Content -Value "${{ secrets.DEPLOYMENT_SSH_KEY }}" -Path ~/.ssh/id_ed25519
+          chmod 0600 ~/.ssh/id_ed25519
+          if(git stash list) {
+              git stash pop
+              install -d -m 0700 ~/.ssh
+              Set-Content -Value "${{ secrets.DEPLOYMENT_SSH_KEY }}" -Path ~/.ssh/id_ed25519
+              chmod 0600 ~/.ssh/id_ed25519
+              git config user.name "ownClouders"
+              git config user.email "devops@owncloud.com"
+              git add translations/
+              git commit -m "[tx] updated client translations from transifex [skip ci]"
+              git push git@github.com:owncloud/ios-app.git
+          }
+
+  normalize-strings:
+    runs-on: macos-13
```

Comment:
@felix-schwarz can we port the script to linux?

Reply:
@TheOneRing It is not possible to port this script to linux, because it runs a Xcode project.

Reply:
Yes but I'm pretty sure https://github.com/owncloud/ios-sdk/blob/master/tools/ocstringstool/ocstringstool/main.swift can also be done in python or bash.

Reply:
It should be possible to also compile & run this tool on Linux without changes, since it only uses Swift Foundation. But I'm not familiar with compiling Swift on Linux.  If you'd like me to look into compiling Swift on Linux, please let me know.

Reply:
@felix-schwarz nothing to do from your side. I am just on it to use Swift on Linux.

Reply:
@TheOneRing Swift normalize strings code is implemented and you can have a look again to the PR